File: src/AS_23_12_week2/wangon/14430.cpp
#include <algorithm>
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main()
{
	int n, m;
	cin >> n >> m;
	vector<vector<int> > arr(n, vector<int>(m));
	vector<vector<int> > dp(n, vector<int>(m));
	for (int i = 0; i < n; i++)
	{
		for (int j = 0; j < m; j++)
			cin >> arr[i][j];
	}
	dp[0][0] = arr[0][0];
	for (int i = 1; i < n; i++)
		dp[i][0] = arr[i][0] + dp[i - 1][0];
	for (int i = 1; i < m; i++)
		dp[0][i] = arr[0][i] + dp[0][i - 1];
	for (int i = 1; i < n; i++)
	{
		for (int j = 1; j < m; j++)
			dp[i][j] = arr[i][j] + max(dp[i - 1][j], dp[i][j - 1]);
	}
	cout << dp[n - 1][m - 1] << endl;
}

File: src/AS_23_12_week2/wangon/14430.py
n, m = map(int, input().split())
arr = [list(map(int, input().split())) for _ in range(n)]
dp = [[0] * m for _ in range(n)]

dp[0][0] = arr[0][0]
for i in range(1, n):
    dp[i][0] = arr[i][0] + dp[i-1][0]
for i in range(1, m):
    dp[0][i] = arr[0][i] + dp[0][i-1]
for i in range(1, n):
    for j in range(1, m):
        dp[i][j] = arr[i][j] + max(dp[i-1][j], dp[i][j-1])
print(dp[n-1][m-1])
